### ISSUE TITLE
BUG: read_parquet doesn't currently work for parquet files with geoarrow columns (from to_arrow)

### ISSUE THREAD

**Issue by bretttully** [geopandas/geopandas#3475]
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of geopandas.
- [x] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

#### Code Sample, a copy-pastable example

Here is a unit test for `test_arrow.py` I was writing to confirm that things were working as expected.

```python
def test_roundtrip_via_arrow_table(tmpdir):
    """Writing to parquet via pyarrow should not raise errors, and it should be
    readable as a GeoDataFrame.
    """
    df = geopandas.GeoDataFrame(
        {"col_str": ["a", "b"], "col_int": [1, 2], "col_float": [0.1, 0.2]},
        geometry=[MultiPolygon([box(0, 0, 1, 1), box(2, 2, 3, 3)]), box(4, 4, 5, 5)],
        crs="EPSG:4326",
    )

    # save the table to a parquet file using pyarrow directly to use the
    # geoarrow implementation
    filename = os.path.join(str(tmpdir), "test.pq")
    table = df.to_arrow()
    if Version(pa.__version__) < Version("14.0.0"):
        orig_table = table._pa_table
    else:
        orig_table = pa.table(table)
    pq.write_table(orig_table, filename)
    assert os.path.exists(filename)

    # read the table back in and convert to a GeoDataFrame and assert equality
    roundtrip_table = pq.read_table(filename)
    df_from_table = GeoDataFrame.from_arrow(roundtrip_table)
    assert_geodataframe_equal(df, df_from_table)

    # read the parquet file directly into a GeoDataFrame and assert equality
    df_from_file = read_parquet(filename)
    assert isinstance(df_from_file, GeoDataFrame)
    assert_geodataframe_equal(df, df_from_file)
```

#### Problem description

Test fails with the following exception at `df_from_file = read_parquet(filename)`

```python
E           ValueError: Missing geo metadata in Parquet/Feather file.
E                       Use pandas.read_parquet/read_feather() instead.
```

**Comment by bretttully** [comment]
I was going to have a go at this, but I'm not quite sure how to do with the bbox filtering... would appreciate a little guidance, and then I can put up a PR.

**Comment by m-richards** [comment]
I can verify this is the case, I think I'm missing something here, because this seems to just be the same as https://github.com/geopandas/geopandas/issues/3406. Is there something else you're pointing out here that is distinct?

I assume for the same reasons as there using `df.to_parquet(filename)` directly isn't an option

**Comment by bretttully** [comment]
The comments on that one suggest that the difference between to_arrow and to_parquet are intentional (the former omitting the geo metadata, in favour of using geoarrow). So that suggests to me that the read_parquet needs to be able to handle both of these cases, which it currently cannot.

Am I misinterpreting the comments on the other ticket?

**Comment by bretttully** [comment]
Maybe I'm missing something in the interoperability between geoarrow and geoparquet here. Perhaps I shouldn't expect geopandas to be able to read a parquet geoarrow file directly?

**Comment by m-richards** [comment]
I think your last sentence is the key point. Current geopandas will only read geoparquets, with the geoparquet metadata, no regular parquets without geometry and no parquets written via arrow which might contain geometry as wkb or geoarrow native encoding with geoarrow metadata.

I guess the question then becomes should we special case for non geoparquet "parquets with geometry" or should we add an option to change how/ what we export converting to arrow. I suspect the former is better since we would want interop with other tools (which means this is indeed distinct from your other issue)

From memory there is overlap on metadata, except for the table level concept of an active geometry columns, but the format is different, nonetheless we should be able to translate even if not round trip entirely.

Have to confess I'm not a geoarrow expert (and I'm writing this away from my computer) so I may not be entirely right. Perhaps @jorisvandenbossche can fill in some details

**Comment by bretttully** [comment]
Yes, interop with other tools creating "parquet with geometry" is exactly what I had in mind here. 

I agree with @jorisvandenbossche justification for not including the geo metadata in `to_arrow` but I do think geopandas should be able to read those same arrow tables from parquet into something sensible (defaulting to first geometry column or whatever logic `from_arrow` is already using, so yes, not a perfect round trip).

**Comment by m-richards** [comment]
Just to round out from me - looking again today, I certainly think we should have a tool for reading these "geometry parquets which aren't geoparquets", and a PR exploring what that would look like is certainly welcome.

I would lean towards not doing this by default / doing it and warning by default as I think we should encourage properly written geoparquets but it would be silly not to make this easier, as I'm sure these kinds of files will exist (also xref #2976 which is a related issue on generalisation).

In terms of your comment about how to deal with the bbox, I would say treat it as omitted given its not part of the geoarrow spec.

Feel free to hold off for now and see if Joris has any input and can be more definitive.

**Comment by theroggy** [comment]
I've never used geoparquet... but possibly using `read_file` will support this case already... The gdal documentation lists some different geometry column types that would be supported: https://gdal.org/en/latest/drivers/vector/parquet.html
